src/features/defi/providers/fox-farming/components/FoxFarmingManager/Deposit/components/Approve.tsx

```diff
@@ -0,0 +1,118 @@
+import { useToast } from '@chakra-ui/react'
+import { ASSET_REFERENCE, toAssetId } from '@shapeshiftoss/caip'
+import { supportsETH } from '@shapeshiftoss/hdwallet-core'
+import { Approve as ReusableApprove } from 'features/defi/components/Approve/Approve'
+import {
+  DefiParams,
+  DefiQueryParams,
+  DefiStep,
+} from 'features/defi/contexts/DefiManagerProvider/DefiCommon'
+import { useFoxFarming } from 'features/defi/providers/fox-farming/hooks/useFoxFarming'
+import { useContext } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { useBrowserRouter } from 'hooks/useBrowserRouter/useBrowserRouter'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import { logger } from 'lib/logger'
+import { poll } from 'lib/poll/poll'
+import { selectAssetById, selectMarketDataById } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { FoxFarmingDepositActionType } from '../DepositCommon'
+import { DepositContext } from '../DepositContext'
+
+type FoxFarmingApproveProps = {
+  onNext: (arg: DefiStep) => void
```

Comment:
You don't need to change it in this PR since we will need to change it in a consistent way (I have a WIP issue for it), but this generic naming is one of the things we should improve after Fox Farming and Idle go in

Reply:
https://github.com/shapeshift/web/issues/2434